error.go

```diff
@@ -56,35 +55,38 @@ type ErrorSource struct {
 
 //MarshalError marshals errors recursively in json format.
 //it can make use of the jsonapi.HTTPError struct
-func (j JSONContentMarshaler) MarshalError(err error) string {
+func (j JSONContentMarshaler) MarshalError(err error) (int, []byte) {
 	httpErr, ok := err.(HTTPError)
 	if ok {
-		return marshalHTTPError(httpErr, j)
+		return httpErr.status, marshalHTTPError(httpErr, j)
 	}
 
 	httpErr = NewHTTPError(err, err.Error(), 500)
 
-	return marshalHTTPError(httpErr, j)
+	return httpErr.status, marshalHTTPError(httpErr, j)
 }
 
 //marshalHTTPError marshals an internal httpError
-func marshalHTTPError(input HTTPError, marshaler ContentMarshaler) string {
+func marshalHTTPError(input HTTPError, marshaler ContentMarshaler) []byte {
 	if len(input.Errors) == 0 {
-		input.Errors = []Error{Error{Title: input.msg, Status: strconv.Itoa(input.status)}}
+		code := ""
+		if input.err != nil {
+			code = input.err.Error()
```

Comment:
so if there is just 1 error the whole error is taken as the `code` field? Should the code not be something explicitly set and in this case, not be used? Becaues it's not explicitly set?